tern/utils/rootfs.py

```diff
@@ -86,8 +85,12 @@ def set_up():
 def extract_layer_tar(layer_tar_path, directory_path):
     '''Assuming all the metadata for an image has been extracted into the
     temp folder, extract the tarfile into the required directory'''
-    with tarfile.open(layer_tar_path) as tar:
-        tar.extractall(directory_path)
+    os.makedirs(directory_path)
+    with open(os.devnull, 'w') as test:
+        result = subprocess.call(['tar', '-tf', layer_tar_path],
```

Comment:
Oh! This is what you meant by security warning. You can run `root_command` here as well and check to see if there is an error or if the result is an empty string.

Reply:
`root_command` will take care of the pipe redirection for you and will return the result and an error message.